packages/trace-viewer/src/traceModel.ts

```diff
@@ -58,6 +59,8 @@ export class TraceModel {
         networkEntry = entry;
       if (entry.filename.includes('src@'))
         this.contextEntry.hasSource = true;
+      if (entry.filename === 'index.html')
+        throw new CustomFetchError('Could not load trace. Did you upload a Playwright HTML report instead? Make sure to extract the archive first and then double-click the index.html file or put it on a web server.');
```

Comment:
This is a wrong place to have this error. Should be somewhere in the UI, call `traceModel.entry('index.html')` or similar.